fl4health/servers/nnunet_server.py

```diff
@@ -177,22 +177,25 @@ def update_before_fit(self, num_rounds: int, timeout: Optional[float]) -> None:
                 information from a client. Defaults to None, which indicates indefinite timeout.
         """
 
+        server_nnunet_plans_exist = self.fl_config.get("nnunet_plans") is not None
+
         # If the per_round_checkpointer has been specified and a state checkpoint exists, we load state
         # NOTE: Inherent assumption that if checkpoint exists for server that it also will exist for client.
         if self.per_round_checkpointer is not None and self.per_round_checkpointer.checkpoint_exists():
             self._load_server_state()
         # Otherwise, we're starting training from "scratch"
-        elif self.per_round_checkpointer is not None:
-            # If the state checkpointer is not None, then we want to do state checkpointing. So we need information
-            # from the clients in the form of get_properties.
+        elif self.per_round_checkpointer is not None or not server_nnunet_plans_exist:
+            # 1) If the per_round_checkpointer is not None, then we want to do state checkpointing. So we need
+            #       information from the clients in the form of get_properties.
+            # 2) If the nnUnet plans are not specified, we also need those plans from the client.
             log(INFO, "")
             log(INFO, "[PRE-INIT]")
             log(INFO, "Requesting properties from one random client via get_properties")
 
-            if self.fl_config.get("nnunet_plans") is None:
+            if not server_nnunet_plans_exist:
                 # If the nnUnet plans are not specified, we also need those plans from the client.
                 log(INFO, "Initialization of global nnunet plans will be sourced from this client")
-            else:
+            if self.per_round_checkpointer is not None:
```

Comment:
The statement below on line 209 is not always True. We're stating that we received the global nnunet plans from the client, when in fact the server may have provided them. Same with the line 211, we may be querying for the properties only for the purpose of the state saving, not for the plans.